### PR TITLE
Collector: add new fields in swagger and check for token in PATCH api

### PR DESCRIPTION
**Pull request by pallavi2209**

### CODE REVIEW

File: api/v1/controllers/collectors.js [hunk 1]
```diff
@@ -138,13 +137,31 @@ function getCollector(req, res, next) {
...
+ * a collector.
  *
  * @param {IncomingMessage} req - The request object
  * @param {ServerResponse} res - The response object
  * @param {Function} next - The next middleware function in the stack
  */
 function patchCollector(req, res, next) {
-  doPatch(req, res, next, helper);
+  let verifyCtrToken = false;
+  for (let i = 0; i < fieldsWritableByCollectorOnly.length; i++) {
```
Comment (iamigo) [inline api/v1/controllers/collectors.js]: maybe something like https://www.npmjs.com/package/just-intersect ...
e.g. 

```
verifyCtrTkn = intersect(Object.keys(req.body), fieldsWritableByCollectorOnly).length > 0;
```

File: api/v1/controllers/collectors.js [hunk 2]
```diff
@@ -34,6 +31,8 @@ const config = require('../../../config');
 const encryptionAlgoForCollector = config.encryptionAlgoForCollector;
 const ZERO = 0;
 
+const fieldsWritableByCollectorOnly = ['osInfo', 'processInfo', 'version'];
```
Comment (shriramshankar) [inline api/v1/controllers/collectors.js]: can we move this to the nouns file?

File: api/v1/controllers/collectors.js [hunk 3]
```diff
@@ -138,13 +137,31 @@ function getCollector(req, res, next) {
...
+    }
+  }
+
+  if (verifyCtrToken) { // verify that token belongs to collector
+    return jwtUtil.verifyCollectorToken(req)
+    .then(() => doPatch(req, res, next, helper))
+    .catch((err) => u.handleError(next, err, helper.modelName));
+  } else {
+    return doPatch(req, res, next, helper);
+  }
```
Comment (shriramshankar) [inline api/v1/controllers/collectors.js]: the else block here is not necessary.

Comment (iamigo) [reply api/v1/controllers/collectors.js]: More specifically, the "else" is not necessary, but what's inside the else is still necessary.

File: api/v1/controllers/collectors.js (hunk 1 above)
Comment (pallavi2209) [reply api/v1/controllers/collectors.js]: How about this: 
`  // verify controller token if atleast one field is writable by collector
  let verifyCtrToken = false;
  const reqBodyKeys = Object.keys(req.body);
  for (let i = 0; i < fieldsWritableByCollectorOnly.length; i++) {
    const fieldName = fieldsWritableByCollectorOnly[i];
    if (reqBodyKeys.indexOf(fieldName) > -1) {
      verifyCtrToken = true;
      break;
    }
  }`

Same complexity and less dependency on third party package. What do you think?

**Comment by pallavi2209** [comment]
@shriramshankar @iamigo updated the PR. @iamigo I added a comment for intersect. Let me know if you still think we should use the package and I will make the changes.

**Pull request merged**